freqtrade/optimize/backtesting.py

```diff
@@ -62,6 +62,8 @@ def __init__(self, config: Dict[str, Any]) -> None:
         LoggingMixin.show_output = False
         self.config = config
         self.results: Dict[str, Any] = {}
+        self.trade_id_counter: int = 0
```

Comment:
@rokups What's the point of having a (semivalid) trade and order id?

it's used to assign "id" - but i don't see where/why we'd need that?

Reply:
Someone on discord said they are using these IDs to store some custom per-trade data and it wasnt working for them in backtesting. I figured why not include that as it costs nothing anyway.

Reply:
it's an anti-pattern to store data by trade-id (at least if you can't recreate it easily - and then why store it in the first place).

and will seem to work UNTIL you restart the bot - and the data is gone :)

Reply:
They were storing it in a dict during runtime so its ok. Besides why would it break in live? IDs are persistent there. And in backtesting we do not need persistence 👀

Reply:
yeah - but the dict info is gone if you restart the bot ... 
the trade however is still there - so you'll need to be VERY careful how you handle this.

Reply:
> so you'll need to be VERY careful how you <insert anything here>.

Fixed that for you :] We are consenting adults, if someone does not understand basics of data lifetime there is no saving them. I see no reason to not allow people who know what they are doing to do what they need to only to make a futile attempt to save someone from themselves. There is a million things they can mess up anyway..